src/main/java/hudson/plugins/active_directory/HttpHeaderFilter.java

```diff
@@ -0,0 +1,127 @@
+import hudson.security.ACL;
+import hudson.security.ACLContext;
+import hudson.util.Scrambler;
+import java.io.IOException;
+import java.util.Collections;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import jenkins.model.Jenkins;
+import jenkins.security.ApiTokenProperty;
+import org.acegisecurity.GrantedAuthority;
+import org.acegisecurity.providers.UsernamePasswordAuthenticationToken;
+import org.acegisecurity.providers.anonymous.AnonymousAuthenticationToken;
+import org.acegisecurity.userdetails.UserDetails;
+import org.acegisecurity.userdetails.UsernameNotFoundException;
+import javax.servlet.Filter;
+import javax.servlet.FilterChain;
+import javax.servlet.FilterConfig;
+import javax.servlet.ServletException;
+import javax.servlet.ServletRequest;
+import javax.servlet.ServletResponse;
+import javax.servlet.http.HttpServletRequest;
+
+class HttpHeaderFilter implements Filter {
+    private static final Logger LOGGER = Logger.getLogger(HttpHeaderFilter.class.getName());
+
+    private final ActiveDirectorySecurityRealm activeDirectorySecurityRealm;
+
+    HttpHeaderFilter(ActiveDirectorySecurityRealm activeDirectorySecurityRealm) {
+        this.activeDirectorySecurityRealm = activeDirectorySecurityRealm;
+    }
+
+    public void doFilter(ServletRequest servletRequest,
+                         ServletResponse response, FilterChain chain)
+            throws IOException, ServletException {
+        HttpServletRequest request = (HttpServletRequest) servletRequest;
+
+        if (Jenkins.getAuthentication() instanceof AnonymousAuthenticationToken) {
+            String authenticatedUserFromApiToken = getUserFromAuthorizationHeader(request);
+
+            String userName = authenticatedUserFromApiToken == null ? getUserFromReverseProxyHeader(request) : authenticatedUserFromApiToken;
+            if (userName != null) {
+                try {
+                    UserDetails userDetails = activeDirectorySecurityRealm.getAuthenticationProvider().loadUserByUsername(userName);
+
+                    GrantedAuthority[] authorities = userDetails.getAuthorities();
+
+                    UsernamePasswordAuthenticationToken auth = new UsernamePasswordAuthenticationToken(userName, "", authorities);
+
+                    try (ACLContext ignore = ACL.as(auth)) {
+                        chain.doFilter(request, response);
+                    }
+                    return;
+                } catch (UsernameNotFoundException e) {
+                    LOGGER.log(Level.FINE, "User from HTTP Header {0} not found in LDAP", userName);
+                }
+            }
+        }
+        chain.doFilter(request, response);
+    }
+
+    String getUserFromAuthorizationHeader(HttpServletRequest request) {
+        String authorization;
+        if ((authorization = request.getHeader("Authorization")) != null && authorization.toLowerCase().startsWith("basic ")) {
```

Comment:
`authorization.toLowerCase(Locale.ROOT)` ... otherwise you will get funky results in turkish